### DISCUSSION TITLE
[Request] require authentication to install or remove tweaks

### DISCUSSION THREAD

**Discussion by FloatingMilkshake** [wstyres/Zebra#808] [Ideas]
It would be nice if we could have Zebra require authentication to install or remove packages (there’s a tweak for Cydia that does this called Cyiometrics by PixelOmer but I don’t believe it works in Zebra). I just stumbled upon how Zebra handles purchases through Packix (and I love it) but I’d personally prefer an option to require authentication to modify free tweaks as well.

**Comment by worldlydev** [comment]
Is this a request for biometric authentication actions in Zebra?

If that's the case then this is totally in my wishlist as well. TouchID/FaceID auth would be great.

Like this tweak: https://www.idownloadblog.com/2019/06/15/installprotect12/
Or the native option in Installer.

**Comment by FloatingMilkshake** [comment]
Yes! This is exactly what I'm looking for. I tried out Installer and saw that it has the option natively, though I don't like Installer's general UI.

That tweak ("InstallProtect12") is the same as Cyiometrics. Cydia requires authentication to modify packages. It'd be so nice if this was native in Zebra.

**Comment by worldlydev** [comment]
> Yes! This is exactly what I'm looking for. I tried out Installer and saw that it has the option natively, though I don't like Installer's general UI.
> 
> That tweak ("InstallProtect12") is the same as Cyiometrics. Cydia requires authentication to modify packages. It'd be so nice if this was native in Zebra.

Ok. Totally agree. This would be amazing. Also agree with the Installer thing. Installer is great but the UI is really off-putting for me for the time being. Can't wait to see what happens in the future when their redesign finally comes as well.

**Comment by worldlydev** [comment]
If implemented with a similar UI to the app store I would totally freak. 
![A2736C02-6827-4782-A43A-09F2781C2DB3](https://user-images.githubusercontent.com/51668635/72175432-6ba4b700-33dc-11ea-9c35-e2cd022bcf93.jpeg)

**Comment by wstyres** [comment]
We can add a simple auth option in 1.1 but anything like the App Store authentication screen (which would be **really** cool) would require some changes to the payment API I believe and would have to wait until 1.2 for the redesign and even then we don’t know how many repos would implement a new in-package manager payment scheme like that.

**Comment by worldlydev** [comment]
> We can add a simple auth option in 1.1 but anything like the App Store authentication screen (which would be **really** cool) would require some changes to the payment API I believe and would have to wait until 1.2 for the redesign and even then we don’t know how many repos would implement a new in-package manager payment scheme like that.

That seems difficult. Why add biometric auth to the payment scheme? I was thinking more just for installing tweaks, removing tweaks, etc. Would that be an easier solution or no?

Your suggestion is awesome but I fear you are right about adoption issues. Perhaps the payment scheme could be worked on later?
